### PR TITLE
SRC blog post

### PR DESCRIPTION
**Pull request by gasche**
cc @maxsnew @dbp 

@dbp is super-busy today and tomorrow and probably won't have time to comment, I plan to wait a few extra days to give him an opportunity for feedback. (please don't merge before that)

### CODE REVIEW

**Comment by bennn** [comment]
I wound up doing an SRC proposal too (using Rosette to do soft contracts). Here's the pdf, later tonight I'll suggest a summary for the blog.
http://www.ccs.neu.edu/home/types/resources/popl2017-src.pdf

**Comment by gasche** [comment]
I volunteer to write the summaries, I like having to read something and describe it in my own words.

I also learned today that @nrioux sent a SRC proposal as well. @nrioux, if you want to be part of the blog post, could you give us the link to your PDF?

(William also had a SRC proposal but he prefers not to put it online yet, which is fine.)

**Comment by dbp** [comment]
Mostly done with presentation now, so I just took a look. 

Overall, I like it (thanks!), but have one issue - the idea is not to extend _both_ source languages - only to extend the one that you are programming with. Which showed up in the example you gave - you have to extend the pure language to talk about impure computations, but you don't actually need to touch the other language. And, correspondingly, you only need a fully abstract compiler from that one language  for the idea to work - a regular old compiler from the other is fine, to the extent that you are only reasoning about equivalences in your (now extended) source language. And, of course, you could have many different languages that you are linking with that have the feature you added (impurity in this case).

**Comment by gasche** [comment]
I won't have time to work on upgrading the descriptions this week -- it's a bit tricky. @dbp and @benn, I release my lock: would you maybe propose summaries that you like? ( @benn: I tried to write something along the lines of, roughly, "Soft contracts work proved that using solvers is viable, but the integration with Racket is not natural, enters Rosette that <has advantage X>", but I don't know Rosette so it's hard to know what to say. ) I would add you to the author list, of course.

**Comment by bennn** [comment]
I'm `bennn` !
Yes I'll try to make a summary

**Comment by maxsnew** [comment]
I haven't uploaded it to my website and it's thanksgiving break now so I
won't get to it until Monday at the earliest.

-Max Stewart New

**Comment by dbp** [comment]
Here's a minimal rewrite of the second paragraph which I think is closer to what I describe. I can't push to this pull request (as it's on @gasche 's fork), but if he approves of the changes he can change!

Daniel's idea is that when a typed language don't have expressive
enough types to safely represent values from another language,
there should be a principled, minimal way to extend the type
system to have enough common ground to talk about the
interaction. If the user's language is pure and the language they
are trying to link with is impure, you extend the user's
language to have both pure and impure arrows. In particular, you
can use those richer types *in source programs* to allow
interaction: if you abstract over a function in the pure
language, but annotate it with the extended "impure function"
type, you make your interaction expectations explicit, and can
safely link with functions in the other language. Formally, you
now have a fully-abstract embedding from the *extended* language
to the multi-language.

**Comment by gasche** [comment]
@dbp thanks, I updated your part. I also removed my summary of Max's work, so there is just his abstract now. If you ultimately think that this would be a better presentation of your submission as well, I can remove the pre-introductoin altogether.

**Comment by maxsnew** [comment]
Here's the link: http://maxsnew.github.io/docs/precision-as-retraction.pdf

**Comment by dbp** [comment]
If the rest are going to be that way, probably. I liked the style of "I'm presenting work that my coworkers are doing", and your third-person introductions (which are a little less formal), but I'm happy either way.

**Comment by bennn** [comment]
I like the "just the abstract" for mine.

**Comment by gasche** [comment]
Okay, let's go no-abstract. I'll add hyperlinks and then it should be fine to go.

(I expected the rant at the end to be more controversial than the summaries but oh well.)

File: blog/_src/posts/2016-11-17-src-submissions.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+> intricacies of either the compiler or the particular language they
+> are linking with.
+
+## Pruning Contracts with Rosette
+
+[Pruning Contracts with Rosette](http://www.ccs.neu.edu/home/types/resources/popl2017-src.pdf)
+Benjamin Greenman
+2016
+
+> Contracts are a pragmatic tool for managing software systems, but
```
Comment (bennn) [inline blog/_src/posts/2016-11-17-src-submissions.md]: Contracts
http://www.ccs.neu.edu/racket/pubs/icfp16-dnff.pdf

**Comment by gasche** [comment]
I included links to Rosette and to the two citations of Max as per @bennn's recommendation. Do you agree this is good to go now?

**Comment by bennn** [comment]
Max also cites `Wadler and Findler`
https://www.eecs.northwestern.edu/~robby/pubs/papers/esop2009-wf.pdf

**Comment by gasche** [comment]
Thanks a lot, fixed.

**Pull request merged**